unit_tests/request_client_server/test_main.cpp

```diff
@@ -532,6 +533,122 @@ TEST_CASE ("Two request_sync_client_server communicate with each other", "[dbg]"
     }
 }
 
+TEST_CASE ("Test response sequence number generator", "")
+{
+  // Test concurrent number generation, that all the generated numbers are unique
+
+  cubcomm::response_sequence_number_generator rsn_gen;
+
+  // Start threads that request numbers from the generator. Save all the numbers.
+  //
+  // In the end, compare all the generated numbers; they should be unique
+  //
+  constexpr size_t THREAD_COUNT = 10;
+  constexpr size_t NUMBER_COUNT = 1000;
+  using numbers_t = std::vector<cubcomm::response_sequence_number>;
+  std::array<std::pair<std::thread, numbers_t>, THREAD_COUNT> threads_info;
+
+  for (auto &ti : threads_info)
+    {
+      numbers_t &numbers = ti.second;
+      numbers.reserve (NUMBER_COUNT);
+      ti.first = std::thread ([&numbers, &rsn_gen] ()
+      {
+	numbers.push_back (rsn_gen.get_unique_number ());
+      }
+			     );
```

Comment:
alignment, but nothing can be done about it

Reply:
fixed